tests/lang_exprs/tmcs_with_erroneous_operand.nim

```diff
@@ -0,0 +1,22 @@
+discard """
+  description: '''
+    Ensure that using the method-call syntax doesn't result in an error when
+    the LHS cannot be typed, but an eligible template/macro with an untyped
+    parameter exists.
+  '''
+  action: compile
+"""
+
+template f(x: untyped) =
+  discard
+
+template f(x, y: untyped) =
+  discard
+
+template `f=`(x, y: untyped) =
+  discard
+
+# must not result in an "undeclared identifier" error
+nonexistentName.f
```

Comment:
This is susceptible to double evaluation of compile time effects? `(static: echo "hi"; nonexistentName).f`

Longer term with both `untyped` overloads being analysed first and perhaps restricting compile time effects/making them transactional maybe that's not an issue? There might be precedence issues with the former. 🤔

Reply:
Yep, it is susceptible to the double evaluation, as well as leakage of symbol table changes (`(var x = 0; nonexistentName).f`).

I figured that, while not ideal, this is an improvement over now, where `(static: ...; nonexistentName).f` doesn't compile at all.